### ISSUE TITLE
Ignore packages or subpackages

### ISSUE THREAD

**Issue by Carpetfizz** [mitmproxy/pdoc#310]
#### Problem Description

My project is a collection of packages that have the following structure

```
package1
    tests_package
    module1
    module2
    ...
package2
    tests_package
    module1
    module2
    ...
```

I would like to ignore `tests_package` from the automatic documentation generation.

#### Proposal

I would like to be able to specify a collection of paths or better yet a search pattern to be ignored by Pydoc.

#### Alternatives

None

#### Additional context

N/A

**Comment by mhils** [comment]
> > Doing so should exclude all unlisted modules. Would that solve your issue?
> 
> Yes, I think this solves the issue in practice but I think it will be easier to have an _exclude list_ instead of keeping track of all the modules that should be included. If the docs are meant to grow with the project, maintaining this list (especially amongst multiple contributors) can become unwieldy.

Yes, I agree on this - if someone wants to give my suggestion in https://github.com/mitmproxy/pdoc/issues/310#issuecomment-954019271 a stab I'd be happy to review a PR :)

**Comment by joubs** [comment]
I implemented something close to what is described in the comment above and submitted a PR. 

The user can now pass a regex that is used to ignore some modules, there is one behavior I'm not entirely happy with though: if the regex matches with the top level module, then the `all_modules` dict is empty at the end of `walk_specs` and an exception is raised. I modified the message to indicate all the modules might have been ignored in that case.
